instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcAttributesExtractor.java

```diff
@@ -29,5 +46,14 @@ public void onEnd(
     if (status != null) {
       attributes.put(SemanticAttributes.RPC_GRPC_STATUS_CODE, status.getCode().value());
     }
+
+    if (requestMetadataValuesToCapture != null) {
+      for (String key : requestMetadataValuesToCapture) {
+        internalSet(
+            attributes,
+            AttributeKey.stringArrayKey(RPC_REQUEST_METADATA_ATTRIBUTE_VALUE_PREFIX + "." + key),
```

Comment:
Consider using a cache for the attribute keys, similar to http: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/07281d05377c7e2de204f37bf9c6293d9664beca/instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/CapturedHttpHeadersUtil.java#L33

Reply:
Good idea, implemented this for gRPC.